### PR TITLE
[master < ] Fix order by memory spike bug

### PR DESCRIPTION
**Pull request by mpintaric55334**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message

Memory spike when using order by has been fixed.

Closes : https://github.com/memgraph/memgraph/issues/1170

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

**Comment by mpintaric55334** [comment]
This needs to be discussed, because it can make problems with other queries that use ORDER BY. cc @antoniofilipovic

File: src/query/interpreter.cpp [hunk 1]
```diff
@@ -234,6 +234,17 @@ std::optional<std::string> GetOptionalStringValue(query::Expression *expression,
   return {};
 };
 
+bool IsOrderByQuery(const std::vector<memgraph::query::Clause *> &clauses) {
+  for (const auto &clause : clauses) {
+    if (clause->GetTypeInfo() == Return::kType) {
+      auto *return_clause = utils::Downcast<Return>(clause);
+      auto isOrderBy = !return_clause->body_.order_by.empty();
+      return isOrderBy;
```
Comment (as51340) [inline src/query/interpreter.cpp:242]:
```suggestion
      return !return_clause->body_.order_by.empty();
```

**Review by as51340 — CHANGES_REQUESTED** [review]
Ping fico for reviewing this I am not relevant enough here. I checked Grafana worklads, nothing seems broken. However, we should add workloads for order by and then inspect what is going on. I can give you jump-start in mgbench tomorrow if you need some help

**Comment by antoniofilipovic** [comment]
I would add workload here, just to be sure how will this behave later on.

**Review by antepusic — COMMENTED** [review]
Good work! I have no comments other than what Andi and FIćo have brought up already – just make sure to address them

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
I think workload needs to land on master first

**Comment by imilinovic** [comment]
I think we had some changes in monotonic memory usage so I think this isn't up to date anymore, closed for now.

**Pull request closed without merging**